docs/source/kedro_project_setup/code_beyond_starter_files.md

```diff
@@ -0,0 +1,133 @@
+# Adding code beyond starter files
+
+After you [create a Kedro project](../get_started/new_project.md) and
+[add a pipeline](../tutorial/create_a_pipeline.md), you notice that Kedro generates a
+few boilerplate files: `nodes.py`, `pipeline.py`, `pipeline_registry.py`...
+
+While those may be sufficient for a small project, they quickly become large, hard to
+read and collaborate on as your codebase grows.
+Those files also sometimes make new users think that Kedro requires code
+to be located only in those starter files, which is not true.
+
+This section elaborates what are the Kedro requirements in terms of organising code
+in files and modules.
+It also provides examples of common scenarios such as sharing utilities between
+pipelines and using Kedro in a monorepo setup.
+
+## Where does Kedro look for code to be located
+
+The only technical constraint for arranging code in the project is that `pipeline_registry.py`
+file must be located in `<your_project>/src/<your_project>` directory, which is where
+it is created by default.
+
+This file must have a `register_pipelines()` function that returns a `tp.Dict[str, Pipeline]`
+mapping from pipeline name to corresponding `Pipeline` object.
+
+Other than that, **Kedro does not impose any constraints on where you should keep files with
+`Pipeline`s, `Node`s, or functions wrapped by `node`**.
+
+```{note}
+You actually can make Kedro look for pipeline registry in a different place by modifying the
+`__main__.py` file of your project, but such advanced customisations are not in scope for this section.
+```
+
+This being the only constraint means that you can, for example:
+* Add `utils.py` file to a pipeline folder and import utilities used by multiple
+  functions in `nodes.py` from there.
+* [Share modules between pipelines](#sharing-modules-between-pipelines).
+  Those could be utility functionalities, or your standalone module responsible for
+  the domain logic of the industry you work at.
+* [Use Kedro in a monorepo setup](#kedro-project-in-a-monorepo-setup) if there are
+  software components independent of Kedro that you want to keep together in the version control system.
+* Delete or rename a default `nodes.py` file, split it into multiple files or modules.
+* Instead of having a single `pipeline.py` in your pipeline folder, split it, for example,
+  into `historical_pipeline.py` and `inference_pipeline.py`.
```

Comment:
I agree that `historical` vs `inference` sound like candidates for leveraging same pipeline just different namespace. To avoid the confusion, I changed the example to this:

```text
If you have multiple large `Pipeline` objects defined in a single `pipeline.py`,
split them into separate `.py` files. For example, in `data_processing` pipeline
you may want to have `cleaning_pipeline.py` and `merging_pipeline.py`.
```

Is that better?

Reply:
sounds good!